client/src/components/LinkField/LinkField.js

```diff
@@ -19,36 +22,63 @@ const section = 'SilverStripe\\LinkField\\Controllers\\LinkFieldController';
  * onChange - callback function passed from JsonField - used to update the underlying <input> form field
  * types - injected by the GraphQL query
  * actions - object of redux actions
+ * isMulti - whether this field handles multiple links or not
  */
-const LinkField = ({ value, onChange, types, actions }) => {
-  const linkID = value;
-  const [typeKey, setTypeKey] = useState('');
+const LinkField = ({ value = null, onChange, types, actions, isMulti = false  }) => {
   const [data, setData] = useState({});
-  const [editing, setEditing] = useState(false);
+  const [editingID, setEditingID] = useState(null);
+
+  // Ensure we have a valid array
+  let linkIDs = value;
+  if (!Array.isArray(linkIDs)) {
+    if (!linkIDs) {
+      linkIDs = [];
+    }
+    if (typeof linkIDs === 'number') {
+      linkIDs = [linkIDs];
+    }
+  }
+
+  // Read data from endpoint and update component state
+  // This happens any time a link is added or removed and triggers a re-render
+  useEffect(() => {
+    if (!editingID && linkIDs.length > 0) {
+      const query = [];
+      for (const id of linkIDs) {
+        query.push(`itemID[]=${id}`);
+      }
+      const endpoint = `${Config.getSection(section).form.linkForm.dataUrl}/?${query.join('&')}`;
+      backend.get(endpoint)
+        .then(response => response.json())
+        .then(responseJson => {
+          setData(responseJson);
+        });
+    }
+  }, [editingID, value && value.length]);
```

Comment:
```suggestion
  }, [editingID]);
```
There's already a conditional to check linkIDs.length within the useEffect block

Reply:
We have to check the value and value length in order to trigger the effect when _new_ links are added. `editingID` is only for editing _existing_ links.

Reply:
OK, set it to `[editingID, linkIDs.length]` since we've converted value to linkIDs

Reply:
That results in an infinite loop, since we're manipulating the value of `linkIDs`.